CANTroller2/src/devices.h

```diff
@@ -596,171 +591,177 @@ class Speedometer : public PulseSensor<float> {
   protected:
     Servo _servo;
-
-    // NOTE: should be marked 'override' but compiler says it doesn't override anything...?
-    void set_native_limits(Param<NATIVE_T> &minParam, Param<NATIVE_T> &maxParam) {
-        this->set_native_limits(minParam, maxParam);
-        _servo.attach(this->_pin, this->min_native->get(), this->max_native->get());
+    virtual void handle_pin_mode() {
+        _servo.writeMicroseconds(this->native.get());  // Write result to servo interface
     }
-
-    void set_human_limits(Param<HUMAN_T> &minParam, Param<HUMAN_T> &maxParam) {
-        this->set_human_limits(minParam, maxParam);
-        _servo.attach(this->_pin, this->min_native->get(), this->max_native->get());
+    virtual void handle_pot_mode() {
+        this->human.set(this->_pot->mapToRange(this->human.get_min(), this->human.get_max())); // map pot to input
+        handle_pin_mode(); // write output
     }
-
   public:
-    ServoPWM(uint8_t pin) : Transducer<NATIVE_T, HUMAN_T>(pin) {
-        _servo.attach(this->_pin);
-    }
+    ServoPWM(uint8_t arg_pin) : Transducer<int32_t, HUMAN_T>(arg_pin) { this->set_can_source(ControllerMode::PIN, true); }
     ServoPWM() = delete;
     void setup() {
         set_pin(this->_pin, OUTPUT);
-    }
-    void write() {
-        _servo.writeMicroseconds((int32_t)this->native.get());  // Write result to servo interface
+        _servo.attach(this->_pin, this->native.get_min(), this->native.get_max());
+        this->set_source(ControllerMode::PIN);
     }
 };
 
 // JagMotor is a class specifically for the brake and steering motors. The jaguar stops the motor when receiving 1500 us pulse,
 //    and varies the speed in one direction if pulse is 1500 to (max~2500) us, the other direction if pulse is 1500 to (min~500) us.
 //    Effectively the difference is these have a center value.
-// class JagMotor : public ServoPWM {
-//   public:
-//     JagMotor (int32_t arg_pin) : ServoPWM(arg_pin) {}
-//     JagMotor() = delete;
-// };
+template<typename HUMAN_T>
+class JagMotor : public ServoPWM<HUMAN_T> {
+  protected:
+    HUMAN_T _center;
+    std::shared_ptr<int32_t> _native_center = std::make_shared<int32_t>();
+    virtual void handle_pin_mode() {
+        int32_t min, max;
+        if (this->human.get() >= _center) {
+            min = static_cast<int32_t>(_center);
+            max = static_cast<int32_t>(this->human.get_max());
+        } else {
+            min = static_cast<int32_t>(this->human.get_min());
+            max = static_cast<int32_t>(_center);
+        }
+        this->native.set(map(static_cast<int32_t>(this->human.get()), min, max, this->native.get_min(), *_native_center));
+        // NOTE: what is this logic for? shouldn't there be something in the #else?
+        #ifdef pwm_jaguars
+            ServoPWM<HUMAN_T>::handle_pin_mode();
+        #else
+            // Send command over serial port
+        #endif
+    }
+  public:
+    JagMotor(int32_t arg_pin) : ServoPWM<HUMAN_T>(arg_pin) {}
```

Comment:
Do we need to use HUMAN_T here for the pin? Are we expecting anything other than int?